### ISSUE TITLE
RPC \net_info reports incorrect `n_peers`

### ISSUE THREAD

**Issue by adizere** [cometbft/cometbft#876]
## Bug Report

Ported from https://github.com/cosmos/gaia/issues/2347. Thanks @gaia for reporting.

> ### Summary of Bug
> More peers than allowed in config.
> 
> ### Version
> 9.0.1
> 
> ## Steps to Reproduce
> How is this possible? Note config.toml last modification date, last time the service was restarted and how many peers it should have VS how many it has.
> 
> ```
> $ cat config.toml | grep _num_ && curl -s http://localhost:26657/net_info | jq -r .result.n_peers && /home/ubuntu/bin/gaiad version && systemctl status gaiad | grep 'Active:' &&
>  ps aux | grep [g]aiad && ll config.toml
> max_num_inbound_peers = 40
> max_num_outbound_peers = 10
> 82
> v9.0.1
>      Active: active (running) since Sat 2023-04-01 19:39:22 UTC; 2h 58min ago
> ubuntu       140  0.0 50.6 9698672 7915472 ?     Ssl  19:39 102:39 /home/ubuntu/bin/gaiad start
> -rw-r--r-- 1 ubuntu ubuntu 19K Mar 20 23:38 config.toml
> ```

### Setup

**CometBFT version** (use `cometbft version` or `git rev-parse --verify HEAD` if installed from source):

Gaia v9.0.1 uses Tendermint Core v0.34.26
https://github.com/cosmos/gaia/blob/4b14265197d9beeb6c8b0d407433c852a5af6422/go.mod#L313

**Have you tried the latest version**: no

**ABCI app** (name for built-in, URL for self-written if it's publicly available): Gaia v9.0.1

**Comment by cason** [comment]
The most likely reason is that described here: https://github.com/cometbft/cometbft/issues/486

**Comment by cason** [comment]
I would rephrase this issue. The problem is not that RPC is providing the wrong numbers, but that the number of peers are likely to not being respecting the configured bounds for inbound, in particular and possibly also for outbound peers.

**Comment by cason** [comment]
It would be helpful if the original question from this issue could be clarified. Namely, whether the number of peers reported by this RPC call matches the number of peers the node actually has at p2p level. This information is periodically logged in INFO level by the following code: https://github.com/cometbft/cometbft/blob/main/p2p/pex/pex_reactor.go#L457

If it is possible to report the latest entries of this log message versus the output of the RPC call, we could understand whether the information reported by the RPC endpoint is correct or not.
